props/saber_fett263_buttons.h

```diff
@@ -381,160 +1813,2794 @@ SaberFett263Buttons() : PropBase() {}
+          if (SetMute(true)) {
+            unmute_on_deactivation_ = true;
+          }
+        }
+        return true;
+
+      case EVENTID(BUTTON_POWER, EVENT_RELEASED, MODE_OFF):
+        if (menu_ && menu_type_ == TRACK_PLAYER) {
+          menu_ = false;
+          menu_type_ = MENU_TOP;
+          return true;
+        } 
+        return false;
+
+      case EVENTID(BUTTON_AUX, EVENT_HELD_LONG, MODE_OFF | BUTTON_POWER):
+#ifdef FETT263_EDIT_MODE_MENU
+        if (!menu_) {
+          if (track_player_) {
+            track_player_->Stop();
+            track_player_.Free();
+          }
+          track_mode_ = PLAYBACK_OFF;
+          if (!SFX_medit) {
+              talkie.Say(talkie_error_in_15, 15);
+              talkie.Say(talkie_font_directory_15, 15);
+              STDOUT.println("Edit Mode prompts missing");
+              return true;
+          }
+          STDOUT.println("Enter Edit Mode");
+          GenerateIniFiles();
+          PlayMenuSound("medit.wav");
+          current_menu_angle_ = fusor.angle2();
+          menu_top_pos_ = 0;
+          menu_ = true;
+          menu_type_ = MENU_TOP;
+          FastOn();
+        } 
+#else
+        GenerateIniFiles();
+        PlayMenuSound("msave.wav");
+#endif
+        return true;
+
+      case EVENTID(BUTTON_NONE, EVENT_TWIST_RIGHT, MODE_OFF | BUTTON_POWER):
+        if (menu_ && menu_type_ == TRACK_PLAYER) {
+          track_mode_ = PLAYBACK_ROTATE;
+          PlayMenuSound("mrotate.wav");
+          return true;
+        }
+        return false;
+
+      case EVENTID(BUTTON_NONE, EVENT_TWIST_LEFT, MODE_OFF | BUTTON_POWER):
+        if (menu_ && menu_type_ == TRACK_PLAYER) {
+          track_mode_ = PLAYBACK_LOOP;
+          PlayMenuSound("mloop.wav");
+          return true;
+        }
+        return false;
+
+      case EVENTID(BUTTON_AUX, EVENT_HELD_LONG, MODE_ON):
+        if (menu_) {
+          PlayMenuSound("mexit.wav");
+          menu_ = false;
+          menu_type_ = MENU_TOP;
+          return true;
+        }
+        if (rehearse_) {
+          PlayMenuSound("mcancel.wav");
+          rehearse_ = false;
+          return true;
+        }
+        if (!battle_mode_) {
+          battle_mode_ = true;
+          if (SFX_bmbegin) {
+            hybrid_font.PlayCommon(&SFX_bmbegin);
+          } else {
+            hybrid_font.DoEffect(EFFECT_FORCE, 0);
+          }
+        } else {
+          battle_mode_ = false;
+          if (SFX_bmend) {
+            hybrid_font.PlayCommon(&SFX_bmend);
+          } else {
+            beeper.Beep(0.5, 3000);
+          }
+        }
+        return true;
+
+      case EVENTID(BUTTON_AUX, EVENT_HELD_LONG, MODE_ON | BUTTON_POWER):
+        if (!menu_ && saved_gesture_control.powerlock || choreo_) {
+          wav_player.Free();
+          choreo_ = false;
+          battle_mode_ = false;
+          Off();
+        }
+        return true;
+
+
+      case EVENTID(BUTTON_NONE, EVENT_TWIST_RIGHT, MODE_ON):
```

Comment:
There seems to be a lot of duplication between EVENT_TWIST_LEFT and EVENT_TWIST_RIGHT.
Also, this is too big to be in here and should be broken out into a separate function anyways.
I'm wondering if we could just make one function called "NavigateMenu(int dir)
From EVENT_TWIST_RIGHT we'll call NavigateMenu(1)
From EVENT_TWIST_LEFT we'll call NavigateMenu(-1)
From my quick reading this seems like it could work, but I haven't checked if it works for everything.

Reply:
I attempted to break out into a function and just call on the event but the menu became very "buggy" missing steps or not responding to events the same as it did when in the actual case and I could not figure out why so I reverted to this set up and have not any issues since.

Reply:
There is no reason for that to happen, so you were doing it wrong somehow.
Were you also trying to put it in a separate class at that time?

Reply:
I spun my wheels in the mud on this, I think maybe some of these breakouts to separate functions might be better if you were able to do how you want and then I test and confirm?  I attempted to break out several pieces of the menu into separate functions within the prop class and some worked fine but others were giving me too much trouble and took me too long to do, at the pace I was going OS6 would be released in 2022.  I propose I finish out the edits that I CAN do and then hand off to you where my abilities lack if that's ok, or I can give you access to edit in my fork directly if you want them edited before merging.  I just seem to have run into some pieces I cannot sort out other ways to do, the approaches in this prop are tested and working perfectly but attempts to move or further dissect were unsuccessful and very time-consuming for me so rather than waste time and create a lot of additional delay your expertise is probably needed.  I plan to finish the edits I can do and then we can discuss the ones I need help on to either have you edit in my fork or merge and break apart later.  I was careful and deliberate in what I had done with limited success and far too much time needed, it will be more efficient if you were able to take some of these over the finish line.

Reply:
these were broken out after some trial and error, working now